.github/workflows/dotnet-build-and-test.yml

```diff
@@ -81,7 +81,7 @@ jobs:
         run: |
           export UT_PROJECTS=$(find ./dotnet -type f -name "*.UnitTests.csproj" | grep -v -E "(Experimental.Orchestration.Flow.UnitTests.csproj|Experimental.Assistants.UnitTests.csproj)" | tr '\n' ' ')
           for project in $UT_PROJECTS; do
-            dotnet test -c ${{ matrix.configuration }} $project --no-build -v Normal --logger trx --collect:"XPlat Code Coverage" --results-directory:"TestResults/Coverage/" -- DataCollectionRunSettings.DataCollectors.DataCollector.Configuration.ExcludeByAttribute=ObsoleteAttribute,GeneratedCodeAttribute,CompilerGeneratedAttribute,ExcludeFromCodeCoverageAttribute
+            dotnet test -c ${{ matrix.configuration }} $project --no-build -v Normal --logger trx --collect:"XPlat Code Coverage" --results-directory:"TestResults/Coverage/" -- DataCollectionRunSettings.DataCollectors.DataCollector.Configuration.ExcludeByAttribute=GeneratedCodeAttribute,CompilerGeneratedAttribute,ExcludeFromCodeCoverageAttribute
```

Comment:
Curious why this change is needed for this PR?

Reply:
`Obsolete` attribute was added to code coverage filtering in recent PRs. As I marked bunch of classes as `Obsolete`, the code coverage decreased, but I think it's incorrect. Even though we marked some classes as `Obsolete`, they are still in-place, and somebody may use it. Which means that even if we do not recommend using these classes anymore, we still need to test them and include them into code coverage, especially if these classes are located in v1 assembly (without `alpha` or `preview` suffix). That's why I removed `Obsolete` attribute from code coverage filtering. Let me know if that makes sense.

Reply:
>  As I marked bunch of classes as Obsolete, the code coverage decreased

You mean the percentage of covered lines/branches decreased? Were the classes you marked as obsolete better covered than the average? I'd have expected the WithData stuff to be, on average, less well covered, and thus I'd have expected marking them obsolete and hiding them from coverage would have actually increased percentage covered.

Reply:
> You mean the percentage of covered lines/branches decreased?

So, without `Obsolete` attribute branch coverage for OpenAI assembly is 80.4%. With `Obsolete` attribute it's 79.5%.

> Were the classes you marked as obsolete better covered than the average?

It could be the case. Which means that we can add more tests to other OpenAI functionality to increase code coverage in more important places. But I'm still not sure if we need to exclude `Obsolete` classes from code coverage report, since deprecated functionality is still in place.